pkg/controller/cainjector/indexers.go

```diff
@@ -92,24 +96,38 @@ func (m *secretMapper) InjectClient(c client.Client) error {
 	return nil
 }
 func (m *secretMapper) Map(obj handler.MapObject) []ctrl.Request {
-	// grab the certificate, if it exists
-	certName := OwningCertForSecret(obj.Object.(*corev1.Secret))
-	if certName == nil {
+	// grab the certificate hash, if it exists
+	hash, ok := obj.Object.(*corev1.Secret).Labels[v1alpha1.CertificateHashKey]
+	if !ok {
 		return nil
 	}
 
 	secretName := types.NamespacedName{Name: obj.Meta.GetName(), Namespace: obj.Meta.GetNamespace()}
-	log := m.log.WithValues("secret", secretName, "certificate", *certName)
+	log := m.log.WithValues("secret", secretName, "certificate hash", hash)
 
-	var cert certmanager.Certificate
-	// confirm that a service owns this cert
-	if err := m.Client.Get(context.Background(), *certName, &cert); err != nil {
-		// TODO(directxman12): check for not found error?
-		log.Error(err, "unable to fetch certificate that owns the secret")
-		return nil
+	// fetch all the certificates from the namespace
+	var certs certmanager.CertificateList
+	if err := m.Client.List(context.Background(), &certs, client.InNamespace(obj.Meta.GetNamespace())); err != nil {
+		log.Error(err, "unable to fetch certificates from secret's namespace")
 	}
 
-	return m.toInjectable(log, m.Client, *certName)
+	// confirm that the certificate corresponding to the hash exists
+	for _, cert := range certs.Items {
+		certName := types.NamespacedName{Name: cert.Name, Namespace: cert.Namespace}
+		cHash, err := controllerutil.HashName(certName)
+		if err != nil {
+			log.Error(err, "unable to calculate hash of certificate name")
```

Comment:
Would it make sense to continue; instead of return? (e.g. we might be lucky and find the desired cert elsewhere in the list)